### ISSUE TITLE
Support overriding function's Dockerfile

### ISSUE THREAD

**Issue by jmkhael** [alexellis/faas-cli#15]
Considering the below mxnet.yml file to build and deploy my function (code residing under a sample/mxnet folder):

```
provider:
  name: faas
  gateway: http://localhost:8080
  network: "func_functions" # this is optional and defaults to func_functions

functions:
  mxnet:
    lang: python
    handler: ./sample/mxnet
    image: jmkhael/faas-mxnet

```
I would like to have a way to tell faas-cli to use a specific Dockerfile (e.g. under sample/mxnet) instead of the default lang template one.

`faas-cli -action build -f ./sample/mxnet/mxnet.yml`

**Comment by alexellis** [comment]
So should the template be used at all in this scenario?

**Comment by jmkhael** [comment]
The idea I had was to apply DRY as much as possible while having the template serve as an initial scaffold.

Ideally I would like to scaffold a FaaS function with a `faas-cli` one liner, and just customize the `FROM image` in this case. 
I thought telling faas-cli to do the default, but maybe change some things so my function has what is needed...

```
functions:
  mxnet:
    lang: python
    from: ./sample/mxnet/Dockerfile.patched
    handler: ./sample/mxnet
    image: jmkhael/faas-mxnet
```

I don't know if this makes any sense

**Comment by alexellis** [comment]
This will be implemented (in kind) via #85 - thanks @jmkhael for a great suggestion.
